smart-contracts/contracts-common/concordium-contracts-common/src/schema.rs

```diff
@@ -1215,6 +1217,21 @@ mod impls {
             Ok(param_schema)
         }
 
+        // Returns an event schema from a versioned module schema
+        pub fn get_event_schema(&self, contract_name: &str) -> Result<Type, VersionedSchemaError> {
+            let versioned_contract_schema = get_versioned_contract_schema(self, contract_name)?;
+
+            let param_event = match versioned_contract_schema {
+                VersionedContractSchema::V0(_) => Err(VersionedSchemaError::NoEventInContract)?,
+                VersionedContractSchema::V1(_) => Err(VersionedSchemaError::NoEventInContract)?,
+                VersionedContractSchema::V2(_) => Err(VersionedSchemaError::NoEventInContract)?,
```

Comment:
For error schemas we have two different errors, 
```
ErrorNotSupported
```
and 
```
NoError
```

We should do the same here, and not coalesce everything.